### PR TITLE
Fix booting error with oracle 11g

### LINKED ISSUE #10896: booting error with oracle 11g
**Which version of ShardingSphere did you use?**
5.0.0-beta
**Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?**
ShardingSphere-JDBC
**Expected behavior**
Correct booting
**Actual behavior**
ORA-00904: "IDENTITY_COLUMN": Invalid identifier
**Reason analyze**
Oracle-11g's ALL_TAB_COLUMNS table does not have IDENTITY_COLUMN field.
source

**Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.**
N/A

### PR DESCRIPTION
**Pull request by xueshiji**
booting error with oracle 11g

Fixes #10896.

Changes proposed in this pull request:
- Fix booting error with oracle 11g

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoader.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
  */
 public final class OracleTableMetaDataLoader implements DialectTableMetaDataLoader {
     
-    private static final String TABLE_META_DATA_SQL = "SELECT OWNER AS TABLE_SCHEMA, TABLE_NAME, COLUMN_NAME, DATA_TYPE, IDENTITY_COLUMN %s FROM ALL_TAB_COLUMNS WHERE OWNER = ?";
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoader.java:47]: @xueshiji This field is added in Oracle 12.1 version, you can refer to the [document](https://docs.oracle.com/database/121/REFRN/GUID-F218205C-7D76-4A83-8691-BFD2AD372B63.htm#REFRN20277).  Therefore, the version judgment logic should be different from the COLLATION field.

**Comment by strongduanmu** [comment]
@xueshiji Please fix ci error first.

**Comment by xueshiji** [comment]
> @xueshiji Please fix ci error first.

I don't know how to fix the ci error. :(

**Comment by strongduanmu** [comment]
> > @xueshiji Please fix ci error first.
> 
> I don't know how to fix the ci error. :(

@xueshiji You can refer to this [document](https://shardingsphere.apache.org/community/cn/contribute/code-conduct/).

**Comment by xueshiji** [comment]
Hi,@strongduanmu ,i have fixed it.

**Comment by strongduanmu** [comment]
> Hi,@strongduanmu ,i have fixed it.

@xueshiji Please execute `mvn clean install -Prelease` locally to ensure CI is passed.

**Comment by xueshiji** [comment]
Hi, @strongduanmu ,Because the sql has been modified, the unit test class also needs to be adjusted, and now it can be built through CI.
Finally, thanks for the guidance

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoaderTest.java [hunk 2]
```diff
@@ -37,106 +37,144 @@
...
+            + " WHERE OWNER = ? AND TABLE_NAME NOT IN ('existed_tbl')";
+
+    private static final String ALL_TAB_COLUMNS_SQL_CONDITION6 = "SELECT OWNER AS TABLE_SCHEMA, TABLE_NAME, COLUMN_NAME, DATA_TYPE  FROM ALL_TAB_COLUMNS"
+            + " WHERE OWNER = ? AND TABLE_NAME NOT IN ('existed_tbl')";
+
     @BeforeClass
     public static void setUp() {
         ShardingSphereServiceLoader.register(DialectTableMetaDataLoader.class);
     }
-
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoaderTest.java:46]: @xueshiji Please keep indents consistent with the previous one.

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoaderTest.java [hunk 3]
```diff
@@ -163,16 +201,16 @@ private ResultSet mockPrimaryKeysMetaDataResultSet() throws SQLException {
         when(result.getString("COLUMN_NAME")).thenReturn("id");
         return result;
     }
-    
+
     private DialectTableMetaDataLoader getTableMetaDataLoader() {
         for (DialectTableMetaDataLoader each : ShardingSphereServiceLoader.getSingletonServiceInstances(DialectTableMetaDataLoader.class)) {
-            if ("Oracle".equals(each.getDatabaseType())) {
+            if ("Oracle" .equals(each.getDatabaseType())) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoaderTest.java:207]: @xueshiji Please delete this useless blank.

**Comment by strongduanmu** [comment]
> Hi, @strongduanmu ,Because the sql has been modified, the unit test class also needs to be adjusted, and now it can be built through CI.
> Finally, thanks for the guidance

@xueshiji Thank you for your efforts. The test case is enough, but there are some minor problems that need to be optimized.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoader.java [hunk 4]
```diff
@@ -105,7 +107,8 @@ private ColumnMetaData loadColumnMetaData(final Map<String, Integer> dataTypeMap
     }
 
     private String getTableMetaDataSQL(final Collection<String> existedTables, final DatabaseMetaData metaData) throws SQLException {
-        String collation = metaData.getDatabaseMajorVersion() >= COLLATION_START_MAJOR_VERSION && metaData.getDatabaseMinorVersion() >= COLLATION_START_MINOR_VERSION ? ", COLLATION" : "";
+        String collation = metaData.getDatabaseMajorVersion() >= COLLATION_START_MAJOR_VERSION ? metaData.getDatabaseMinorVersion() >= COLLATION_START_MINOR_VERSION ? ", IDENTITY_COLUMN, COLLATION"
```
[the remark this answers is not in the document]
Comment (xueshiji) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/OracleTableMetaDataLoader.java]: @strongduanmu  What do you think of it now？

**Review by strongduanmu — APPROVED** [review]

**Comment by strongduanmu** [comment]
@xueshiji Thank you for your contribution and look forward to seeing you in other issues. 👍

**Pull request merged**